Cython/Compiler/Nodes.py

```diff
@@ -1606,6 +1606,79 @@ def analyse_declarations(self, env, enum_entry):
             enum_entry.type.values.append(entry.name)
 
 
+class CppScopedEnumDefNode(StatNode):
```

Comment:
This (and `CppScopedEnumDefItemNode`) look relatively similar to their equivalent regular enums. My personal inclination would be to try to do it all in the existing classes, maybe with an optional `scope` parameter. Or if there's reasons not to do that then use inheritance to capture the shared bits.

There may be good reasons I'm not seeing why this wouldn't work though.

Reply:
Thanks and 100% agreed about the code repetition here. I'm more inclined to pursue the inheritance route. The two constructs `enum` and `enum class` have different enough behaviour that I think trying to implement them both with the same class could become a hindrance in time.

Reply:
@da-woods I'm having some second thoughts here and would love your input:

`CEnumType` inherits from `CIntLike`, and rightly so, as C-style enums are trivially convertible to integers: https://wandbox.org/permlink/b7kkoNz4gI2mjNXx

However, scoped enums are *not* trivially convertible to integers: https://wandbox.org/permlink/2CB5lmrX04rokcT6. So implicit treatment of a scoped enum as an int would likely break things.

As I mentioned [here](https://github.com/cython/cython/pull/3640#discussion_r430664189), I think the right semantics could be captured with something like a `BaseEnum` class, and subclasses `CEnum` and `CppScopedEnum`.

Does that sound reasonable?

Reply:
Ah apologies, I see that this comment if about the associated `Node`, not `PyrexType`.

I think inheritance could still be applied here.